codeworld-base/src/Internal/Num.hs

```diff
@@ -128,6 +128,12 @@ instance P.Show Number where
         stripZeros =
             P.reverse . P.dropWhile (== '.') . P.dropWhile (== '0') . P.reverse
 
+instance P.ShowShortFloat x where
```

Comment:
The idea was to copy the `show` function from this instance, but not to define a new instance of your own.  An instance would need to be an instance of some type class, and there's no class called `ShowShortFloat` to have an instance of.

I think you just want a plain function, like this:

    showShortFloat :: Double -> String
    showShortFloat x = stripZeros (showFFloatAlt (Just 4) x "")
      where stripZeros = reverse . dropWhile (== '.') . dropWhile (== '0') . reverse

You want this in CodeWorld.Driver, not here.  This file doesn't need to change.

Reply:
Thanks. I misunderstood. It would still be nice to know how to go up and down the file structure also using import.

Reply:
I'm not sure I understand what you're asking.  There are two different packages here: `codeworld-api` and `codeworld-base`.  You can't import stuff from `codeworld-base` into `codeworld-api`, because codeworld-api doesn't depend on codeworld-base.  The dependencies of codeworld-api are listed in https://github.com/google/codeworld/blob/master/codeworld-api/codeworld-api.cabal  But you also can't make `codeworld-api` depend on `codeworld-base`, because `codeworld-base` depends on `codeworld-api` already.  If you added the other direction, you'd have a dependency cycles, and neither package could be built because the other one needs to be built first.

This is precisely why I was suggesting that you copy the code here over to where you need it, instead of trying to reuse one implementation in both places.  There simply isn't a good place yet to put common utility code that's shared by both `codeworld-api` and `codeworld-base`.

Reply:
I see. Makes sense. Thanks!